static/js/publisher/release/selectors/index.js

```diff
@@ -261,3 +261,29 @@ export function getRevisionsFromBuild(state, buildId) {
     revision => getBuildId(revision) === buildId
   );
 }
+
+// return the progressive release status based on channel, arch
+export function getProgressiveState(state, channel, arch, revision) {
+  const { releases } = state;
+  const allReleases = releases.filter(
+    item => channel === getChannelString(item) && arch === item.architecture
+  );
+
+  const releaseIndex = allReleases.findIndex(
+    item => revision === item.revision
+  );
+
+  let progressive = null;
+
+  const release = allReleases[releaseIndex];
+
+  if (release && release.progressive && release.progressive.key) {
+    progressive = release.progressive;
+
+    if (allReleases[releaseIndex + 1]) {
```

Comment:
We need to verify if that's correct (if there can be multiple progressive releases with same revisoin, or do they override each other).

Maybe add a TODO so we remember about it.